platform/lib/routers/surveyComponent.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Joi = require('joi');
+const express = require('express');
+const log = require('@lib/utils/log')('Survey Component Response');
+const credentials = require('@lib/utils/credentials');
+const {GoogleSpreadsheet} = require('google-spreadsheet');
+const surveyEndpoint = '/fez-survey-response';
+let Doc = undefined;
+let SURVEY_RESPONSE_SHEET_ID;
+let GOOGLE_SERVICE_ACCOUNT_EMAIL;
+let GOOGLE_PRIVATE_KEY;
+
+const schema = Joi.object({
+  survey: Joi.string().required().min(1),
+  questions: Joi.when('dismissed', {
+    is: false,
+    then: Joi.array()
+      .items(
+        Joi.object({
+          text: Joi.string().min(1).required(),
+          answer: Joi.alternatives()
+            .try(
+              Joi.string().min(1).max(200),
+              Joi.array().items(Joi.string().min(1).max(200)),
+              null
+            )
+            .required(),
+        })
+      )
+      .required()
+      .min(1),
+  }),
+  dismissed: Joi.boolean().optional(),
+  url: Joi.string().uri().required(),
+  shownAt: Joi.string().isoDate().required(),
+  originalText: Joi.string().optional(),
+});
+
+const schemaOptions = {
+  abortEarly: false,
+  stripUnknown: true,
+};
+
+function validateRequest(req, res, next) {
+  if (req.method !== 'POST') {
+    return res.sendStatus(405);
+  }
+
+  const {error, value} = schema.validate(req.body, schemaOptions);
+
+  if (error) {
+    res
+      .status(422)
+      .send(
+        `Validation error: ${error.details.map((x) => x.message).join(', ')}`
+      );
+  } else {
+    req.body = value;
+    next();
+  }
+}
+
+async function getDoc() {
+  if (Doc === undefined) {
+    SURVEY_RESPONSE_SHEET_ID = await credentials.get(
+      'SURVEY_RESPONSE_SHEET_ID'
+    );
+    GOOGLE_SERVICE_ACCOUNT_EMAIL = await credentials.get(
+      'GOOGLE_SERVICE_ACCOUNT_EMAIL'
+    );
+    GOOGLE_PRIVATE_KEY = await credentials.get('GOOGLE_PRIVATE_KEY');
+
+    Doc = new GoogleSpreadsheet(SURVEY_RESPONSE_SHEET_ID);
+
+    await Doc.useServiceAccountAuth({
+      client_email: GOOGLE_SERVICE_ACCOUNT_EMAIL,
+      private_key: GOOGLE_PRIVATE_KEY,
+    });
+
+    await Doc.loadInfo();
+  }
+
+  return Doc;
+}
+
+async function uploadAnswer(req, res) {
+  try {
+    const survey = req.body;
+    const doc = await getDoc();
+    let sheet = doc.sheetsByTitle[survey.survey];
```

Comment:
s/sheet/surveyResponse

Reply:
suggestion: extract into a method `getOrCreateSheet(survey)`

Reply:
to your first point - line 107 refers to the sheet within a google sheet, not the survey response (i.e. the page collecting all responses). Do you believe `surveyResponse` is a better mental model for it?

Reply:
Sorry, messed up my comment. What I meant was:

s/sheet/surveySheet
s/survey/surveyResponse

Does that make more sense?

Reply:
yep! thank you

Reply:
updated